### DISCUSSION TITLE
Plans to move to node-fetch 3

### DISCUSSION THREAD

**Discussion by DarthHater** [mswjs/msw#1183] [General]
Hi!

I was recently upgrading a project to use `node-fetch` 3.x and we use `msw`, which obviously is on `2.6.7`. I forked `msw` and started playing around with upgrading to `node-fetch` 3, but it hurt my brain a tiny bit for one, and for two, I was like, hmmm, do they even want to do that? Figured I'd come here and ask a question! If you'd like me to open an issue or something else, let me know!

Cheers,
Jeffry

**Comment by kettanaito** [comment]
Hey, @DarthHater. 

Please, do you mean migrating to `node-fetch` _internally_ in the library or adding support for intercepting requests made via `node-fetch@3`? 

If it's the latter, then I believe it's already supported. The underlying request logic of `node-fetch` won't change between releases as it relies on the `http` module.

If you're speaking about the library version of `node-fetch`:

https://github.com/mswjs/msw/blob/9a21072554ba5a06f268f7caa6f645dabe3d1f88/package.json#L85

Then could you please share more details what errors you've encountered after the upgrade? I think opening a work-in-progress pull request would be the best. We could collaborate on it together.

**Comment by DarthHater** [comment]
The idea was to upgrade `node-fetch` as a dependency.

Specifically, between 2 and 3, your `require` of `node-fetch` quits working, because in 3 they switched to ESM and the commonjs style import no longer works! I went in to futz around with this locally and just started going oh lol doing a dynamic import isn't gonna be as easy as I thought it would be, hah.

Glad to throw up a PR and glad to work together!

**Reply by DarthHater** [reply]
FTR the reason I tried doing this is because a colleague was attempting to upgrade `node-fetch` in general on a `next` project, and we use `msw` to accomplish some of our `playwright` testing, but when I went super nuclear and just told `yarn` to resolve `node-fetch` to 3.x (using `resolutions`), `msw` was notttt happy :)

**Reply by kettanaito** [reply]
I'd be happy to look at the pull request! Perhaps, it's worth keeping in mind that `node-fetch` doesn't lock up in from the consumer's standpoint. It's our internal dependency and we use it to provision requests in Node.js. This does not affect what versions of `node-fetch` you may be using in your application (so by all means do update). 

> I went super nuclear and just told yarn to resolve node-fetch to 3.x

I think that's the issue. Forcing major version bumps in fourth-party dependencies is a risky action. I'd advise against doing that. You can update your app to `node-fetch@3` but let MSW and other tooling still use `node-fetch@2`.
